Update Entry: General IDE (#937)
Codecademy/docs: content/general/concepts/ide/ide.md

  - 'paths/front-end-engineer-career-path'
---

> An **IDE**, or Integrated Development Environment, enables programmers to consolidate the different aspects of writing a computer program.

Reviewer: ```suggestion
An **Integrated Development Environment (IDE)**, enables programmers to consolidate the different aspects of writing a computer program.
```

> IDEs increase programmer productivity by combining common activities of writing software into a single application: editing source code, building executables, and debugging.

Reviewer: ```suggestion
IDEs increase programmer productivity by featuring common activities of writing software into a single application..
```

> 

Reviewer: For the header on the next line, I would suggest renaming it to "Common Features". Then, I would add a hashtag `#` to the following headers:

- ~Autocomplete~ (change this to "Autocompletion")
- Building Executables
- Debugging

- [Microsoft Visual Studio](https://visualstudio.microsoft.com)

The IDE you decide to use depends on what area of computer programming you wish to pursue. Some IDEs, like Android Studio and Xcode, are dedicated solely for building mobile applications.

### Online IDEs

There are many online IDEs that allow users to practice with snippets of code without installing any software on their local machine. Most provide support for several different languages.

> One example is Codecademy's own [Codebyte Editor](https://www.codecademy.com/codebyte-editor). It allows experimenting with [C++](https://www.codecademy.com/resources/docs/cpp), C#, [Go](https://www.codecademy.com/resources/docs/go), [JavaScript](https://www.codecademy.com/resources/docs/javascript), [PHP](https://www.codecademy.com/resources/docs/php), [Python 3](https://www.codecademy.com/resources/docs/python), [Ruby](https://www.codecademy.com/resources/docs/ruby) and Scheme.

Reviewer: ```suggestion
One example is Codecademy's own [Codebyte Editor](https://www.codecademy.com/codebyte-editor). It allows experimenting with [C++](https://www.codecademy.com/resources/docs/cpp), C#, [Go](https://www.codecademy.com/resources/docs/go), [JavaScript](https://www.codecademy.com/resources/docs/javascript), [PHP](https://www.codecademy.com/resources/docs/php), [Python 3](https://www.codecademy.com/resources/docs/python), [Ruby](https://www.codecademy.com/resources/docs/ruby), and Scheme.
```

> Another example is [JDoodle](https://www.jdoodle.com/) which offers support for over 75 languages including [Java](https://www.codecademy.com/resources/docs/java), [Python 2](https://www.codecademy.com/resources/docs/python), [R](https://www.codecademy.com/resources/docs/r) and [Swift](https://www.codecademy.com/resources/docs/swift).

Reviewer: ```suggestion
Another example is [JDoodle](https://www.jdoodle.com/) which offers support for over 75 languages including [Java](https://www.codecademy.com/resources/docs/java), [Python 2](https://www.codecademy.com/resources/docs/python), [R](https://www.codecademy.com/resources/docs/r), and [Swift](https://www.codecademy.com/resources/docs/swift).
```
